webview/src/plots/hooks/useSetupSmoothPlot.ts

```diff
@@ -0,0 +1,50 @@
+import { useDispatch, useSelector } from 'react-redux'
+import { View } from 'react-vega'
+import { setSmoothPlotValue } from '../components/templatePlots/templatePlotsSlice'
+import { PlotsState } from '../store'
+
+interface VegaState {
+  signals?: { [name: string]: number | undefined }
+  data?: unknown
+}
+
+export const useSetupSmoothPlot = (id: string, isZoomedIn = false) => {
+  const dispatch = useDispatch()
+  const smoothPlotValues = useSelector(
+    (state: PlotsState) => state.template.smoothPlotValues
+  )
+
+  return (view: View) => {
+    const state = view.getState() as VegaState
+    const defaultValue = smoothPlotValues[id]
+
+    if (!state?.signals?.smooth) {
+      return
+    }
+
+    if (defaultValue) {
+      view.setState({
+        ...state,
+        signals: { ...state.signals, smooth: defaultValue }
+      })
+    }
+
+    if (isZoomedIn) {
```

Comment:
I couldn't think of a way to update the zoomable plots' values when the value changes inside of a plot model. 

We could save the `view` into a reference so we could call `setState` later but I'm not sure how we could detect when the zoomable plots need an update. I disabled running `setSmoothPlotValue` on model slider change for now but any ideas?

Reply:
what is the difference between updating in the modal and non-modal states?

Reply:
> what is the difference between updating in the modal and non-modal states?

We're updating the state `onViewReady` aka when the plot first renders. Users' always see the modal on first render so it's easy to update the state there unlike the non-modals which stay on the page.

Though we might be able to detect when the state needs to be updated even after first render by tracking the `smoothPlotValue` and holding the `view` in a reference so we can check for differences and update if they're different 🤔

Reply:
> Though we might be able to detect when the state needs to be updated even after first render by tracking the smoothPlotValue and holding the view in a reference so we can check for differences and update if they're different 🤔

Modals state is now corrected!